src/ucp/stream/stream_recv.c

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2017.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include <ucp/core/ucp_ep.h>
+#include <ucp/core/ucp_worker.h>
+#include <ucp/core/ucp_context.h>
+#include <ucp/core/ucp_request.h>
+
+#include <ucs/datastruct/mpool.inl>
+#include <ucs/debug/profile.h>
+
+#include <ucp/tag/eager.h> /* TODO: move ucp_eager_sync_hdr_t to common file */
```

Comment:
can we avoid using ucp_eager_sync_hdr_t ?

Reply:
It may require to refactor common rdesc usage. I would not like to do it in this PR

Reply:
can you pls check if possible w/o changing too much common code?

Reply:
i'll see